Index out of range in Kubelet source

```
panic: runtime error: index out of range

goroutine 2302 [running]:
k8s.io/heapster/metrics/sources/kubelet.(*kubeletMetricsSource).decodeMetrics(0xc820747260, 0xc820e8e900, 0x0, 0x0, 0xc821107260)
	/usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/metrics/sources/kubelet/kubelet.go:154 +0x18cf
k8s.io/heapster/metrics/sources/kubelet.(*kubeletMetricsSource).ScrapeMetrics(0xc820747260, 0xece3d3d6c, 0xc800000000, 0x1983580, 0xece3d3d8a, 0xc800000000, 0x1983580, 0xc820b52000)
	/usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/metrics/sources/kubelet/kubelet.go:237 +0x4c8
k8s.io/heapster/metrics/sources.scrape(0x7f272c6b6268, 0xc820747260, 0xece3d3d6c, 0x0, 0x1983580, 0xece3d3d8a, 0xc800000000, 0x1983580, 0x0)
	/usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/metrics/sources/manager.go:156 +0x328
k8s.io/heapster/metrics/sources.(*sourceManager).ScrapeMetrics.func1(0x7f272c6b6268, 0xc820747260, 0xc82100cde0, 0xece3d3d6c, 0x0, 0x1983580, 0xece3d3d8a, 0x0, 0x1983580, 0xece3d3da3, ...)
	/usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/metrics/sources/manager.go:86 +0x15f
created by k8s.io/heapster/metrics/sources.(*sourceManager).ScrapeMetrics
	/usr/local/google/home/pszczesniak/go/src/k8s.io/heapster/metrics/sources/manager.go:102 +0x47a
```

--- a/metrics/sources/kubelet/kubelet.go
+++ b/metrics/sources/kubelet/kubelet.go
@@ -83,6 +83,10 @@ func (this *kubeletMetricsSource) String() string {
 }
 
 func (this *kubeletMetricsSource) decodeMetrics(c *cadvisor.ContainerInfo) (string, *MetricSet) {
+	if len(c.Stats) == 0 {
+		return "", nil
+	}
+
 	var metricSetKey string
 	cMetrics := &MetricSet{
 		MetricValues: map[string]MetricValue{},
@@ -235,7 +239,7 @@ func (this *kubeletMetricsSource) ScrapeMetrics(start, end time.Time) *DataBatch
 	keys := make(map[string]bool)
 	for _, c := range containers {
 		name, metrics := this.decodeMetrics(&c)
-		if name == "" {
+		if name == "" || metrics == nil {
 			continue
 		}
 		result.MetricSets[name] = metrics

